dynamorm/model.py

```diff
@@ -82,27 +82,21 @@ def should_transform(inner_class):
                 from .types._marshmallow import Schema
             elif "schematics" in sys.modules:
                 from .types._schematics import Schema
-
-                # Pull all of our fields up onto the main schema, obeying MRO
-                # This is done to ensure that "mixin" class fields get properly declared for Schematics
-                def pull_up_fields(cls):
-                    for base in reversed(cls.__bases__):
-                        for k, v in six.iteritems(base.__dict__):
-                            if isinstance(v, Schema.base_field_type()):
-                                setattr(attrs["Schema"], k, v)
-                        pull_up_fields(base)
-
-                pull_up_fields(attrs["Schema"])
             else:
                 raise DynaModelException(
                     "Unknown Schema definitions, we couldn't find any supported fields/types"
                 )
 
-            SchemaClass = type(
-                "{name}Schema".format(name=name),
-                (Schema,) + attrs["Schema"].__bases__,
-                dict(attrs["Schema"].__dict__),
-            )
+            if issubclass(attrs["Schema"], Schema.base_schema_type()):
+                SchemaClass = type(
+                    "{name}Schema".format(name=name), (Schema, attrs["Schema"]), {},
+                )
+            else:
+                SchemaClass = type(
+                    "{name}Schema".format(name=name),
+                    (Schema,) + attrs["Schema"].__bases__,
+                    dict(attrs["Schema"].__dict__),
+                )
```

Comment:
If we're asking users to be explicit, why not just say they _always_ have to inherit from the `base_schema_type` even if they otherwise have no inheritance? It'll ensure this code can be removed as well, as well as the `should_transform` brittleness above?